src/middleware/navigation/check.condition.ts

```diff
@@ -65,7 +66,15 @@ export const checkBOsOrMOsDetailsEntered = (appData: ApplicationData): boolean =
 // UPDATE journey
 
 export const checkOverseasEntityNumberEntered = (appData: ApplicationData): boolean => {
-  return checkHasAppData(appData) && (appData[OeNumberKey] || "").length !== 0;
+  return checkHasAppData(appData) && (appData[EntityNumberKey] || "").length !== 0;
+};
+
+export const checkHasOverseasEntity = (appData: ApplicationData): boolean => {
+  return checkHasAppData(appData) && appData[EntityKey] !== undefined;
+};
+
+export const checkHasDateOfCreation = (appData: ApplicationData): boolean => {
+  return checkHasAppData(appData) && appData[UpdateKey]?.date_of_creation !== undefined;
 };
```

Comment:
I think the first condition of each of these should be calling the previous check like the register ones?
e.g. should `checkOverseasEntityNumberEntered`'s first condition to `checkHasOverseasEntity` rather than `checkHasAppData`

Reply:
This check is combined (using &&) with checkHasEntity in the middleware so that is checking the previous page looked up the entity & date of creation.

Reply:
Cool, sorry maybe left comment in wrong place .. what about the other one, `checkOverseasEntityNumberEntered`, is that ok? Should it be updated to the following now you have added new `checkHasOverseasEntity`:

```
export const checkOverseasEntityNumberEntered = (appData: ApplicationData): boolean => {
  return checkHasOverseasEntity(appData) && (appData[EntityNumberKey] || "").length !== 0;
};
```

Reply:
No, getting the entity requires the number. The hasOverseasEntity check is just a more detailed check for the confirmation page (checks entity was loaded) with others then checking that the entity number is present. It's probably more than is needed as you now can't get forward from the query page unless the entity is found (error handling ensures this). But I've added an explicit check for the entity number to the hasOverseasEntity validation (and also added some more simple check tests).